### PR TITLE
Run3 tt+X cards (Madgraph5_aMC@NLO)

### PR DESCRIPTION
**Pull request by Cvico**
Dear experts,

I'm a MC contact for the TOP group. I've created this new PR for the merging of
some Madgraph5_aMC@NLO cards. In particular for: TTTT, TTWW, TTZZ and TTWZ 
processes.

This is based on a previous PR I sent with additionally TTW, TTZ and TTG processes (which
I now have removed given that these will be centrally produced). I'd like to know if these cards
are valid as they are, or if I have to modify anything before.

I've closed the [other PR ](https://github.com/cms-sw/genproductions/pull/3262) and opened this new one
because I think I accidentaly messed it up when pushing to my fork. This PR now only includes my modifications
to the cards stored in `bin/MadGraph5_aMCatNLO/cards/production/13p6TeV`. Sorry for the possible inconveniences.

Best regards,
Carlos Vico

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTTT_5f_NLO/TTTT_5f_NLO_customizecards.dat [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#put card customizations here (change top and higgs mass for example)
+set param_card mass 6 173.0
+set param_card yukawa 6 173.0
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTTT_5f_NLO/TTTT_5f_NLO_customizecards.dat:3]: we use 172.5 for UL, and for Run3 TTbar, we also use 172.5, so i suggest them to be consistent

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTWW_5f_LO/TTWW_5f_LO_customizecards.dat [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#put card customizations here (change top and higgs mass for example)
+set param_card mass 6 173.0
+set param_card yukawa 6 173.0
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTWW_5f_LO/TTWW_5f_LO_customizecards.dat:3]: same comment here

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTWZ_5f_LO/TTWZ_5f_LO_customizecards.dat [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#put card customizations here (change top and higgs mass for example)
+set param_card mass 6 173.0
+set param_card yukawa 6 173.0
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTWZ_5f_LO/TTWZ_5f_LO_customizecards.dat:3]: same comment here

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTZZ_5f_LO/TTZZ_5f_LO_customizecards.dat [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#put card customizations here (change top and higgs mass for example)
+set param_card mass 6 173.0
+set param_card yukawa 6 173.0
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTZZ_5f_LO/TTZZ_5f_LO_customizecards.dat:3]: same comment here

**Comment by menglu21** [comment]
Another comment is that for ttWW ttWZ, it may be not trivial to generate NLO sample, but for ttZZ it should be straightforward to extend it to NLO, maybe you can have a test on that.

**Comment by Cvico** [comment]
Hi Meng,
Thank you for the review. For the parameters, I'll change them in the next iteration, thanks.

As for TTVV, do you want me to make an specific test to check out if this is indeed straightforward? I understand that for TTW* this is not straightforward because the polarization effects of the W over the final states, right?

**Comment by Cvico** [comment]
Hi @menglu21, hope you have restful holidays after our last iteration!

I have updated the PR with the customizedCards parameters change to Run3 standards for the TTTT NLO and TTVV (V = W,Z) LO processes alone.

Regarding your suggestion of TTZZ at NLO, I have been trying to find a good way of validating that this indeed yields any issues. So far I've created the cards for the NLO (I did not push this into my repository yet) and run the gridpack as always. In principle there's no issue while creating the gridpack, and I get the cross sections:

     Cross-section :        0.001567 +- 6.742e-06 pb [LO]
     Total cross section: 0.00211   +- 3.0e-06     pb [NLO]
     
I'd like to ask you if you have in mind any specific check on how to validate this sample maybe at some distribution level. I've been looking to do this, but since I'm not 100% sure why did you suggest to go with NLO only for ttZ, maybe you had in mind something I can take a look into.

**Comment by menglu21** [comment]
Hi @Cvico thanks for the update and sorry for the late reply. My concerning is about the overlap with other top processes since you are using 5 flavor scheme and there could be additional "b" emission (with "w" to be decayed from another top). I'm fine with the LO process, just want to mention the NLO may be interesting to check. The distributions you can look could be jet kinematics at LHE-level or at GEN-level, and the cross section between NLO and LO. Another comment is on the madspin, you use madspin for NLO TTTT, for the others you don't specify the decay, do you plan to specify the decay through pythia during production

**Comment by Cvico** [comment]
Hi,

I see, then we would to have some kind of Diagram removal/Diagram subtraction scheme as it is done for tW production? (I don't exactly know how these are applied for tW channel, but I guess this is a similar problem as in ttbar and tW interference right?.

Regarding the decays, I just adapted the card from their latest versions used in Run2 legacy and UL: LO production where the decays are handled by the shower for the ttVV, and NLO+Madspin for TTTT. So in principle my plan was indeed to specify decay chains through pythia; although I could specify the madspin decays also for the ttVV process since it should be straightforward to include.

**Comment by Cvico** [comment]
Hi,

Sorry I made an error while preparing another PR to this same repository. I wanted to have
the cards for ST production using aMC@NLO approved. I was going to do that in another PR,
but mistook the branch I was working on and upload them into this one... How should I proceed?

**Comment by Saptaparna** [comment]
@Cvico, did you want to update the cards to address @menglu21 concerns regarding the 5 fs vs. 4 fs? If not, I think we can merge.

**Comment by Cvico** [comment]
Hi, @Saptaparna. Thanks for commenting! So from TMG side, we think that it would be more useful to have the LO cards merged in the meantime, and put the ttVV at NLO (Where the issue in the 5fs and 4fs appears) on hold for the moment, and maybe add the cards for those in a later pull request.  After all, LO accuracy is probably sufficient for many analysis in these early stages of Run3, since with our current luminosity we are most likely not very sensitive to these NLO effects yet (ttVV has ~0.002 pb cross section).

**Comment by menglu21** [comment]
> Hi, @Saptaparna. Thanks for commenting! So from TMG side, we think that it would be more useful to have the LO cards merged in the meantime, and put the ttVV at NLO (Where the issue in the 5fs and 4fs appears) on hold for the moment, and maybe add the cards for those in a later pull request. After all, LO accuracy is probably sufficient for many analysis in these early stages of Run3, since with our current luminosity we are most likely not very sensitive to these NLO effects yet (ttVV has ~0.002 pb cross section).

Hi @Cvico I'm fine that you can keep the LO currently. you may withdraw your new added content and we can merge the initial one.

**Comment by Cvico** [comment]
Hi, I've removed the changes from the commit on the tX samples. However I'm getting an error because merge conflicts
with TTWW and TTZZ, because it seems these cards are already merged in the central repository... 

https://github.com/cms-sw/genproductions/tree/master/bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTWW_5f_LO

These were merged on Jan13, and my PR is from Dec 10.

**Comment by menglu21** [comment]
> Hi, I've removed the changes from the commit on the tX samples. However I'm getting an error because merge conflicts with TTWW and TTZZ, because it seems these cards are already merged in the central repository...
> 
> https://github.com/cms-sw/genproductions/tree/master/bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/TTWW_5f_LO
> 
> These were merged on Jan13, and my PR is from Dec 12.

are those cards the same, if yes, you could remove them from your PR and we can merge your PR

**Comment by Cvico** [comment]
Hi, @menglu21 

I apologise because I've just realised this PR has been on hold for a very long time. 
I've implemented the change you asked me to do after checking that the ttWW and ttZZ cards I was trying
to push were the same as the ones that are already merged in the master branch.  

So at the end we only have to push ttWZ and TTTT.

**Comment by Cvico** [comment]
Hi, If everything is OK now, would it be possible to merge these ttWZ and tttt cards?

**Pull request merged**